### ISSUE TITLE
v0.12.4 Uncaught TypeError: Cannot read property 'lineVertex' of null

### ISSUE THREAD

**Issue by gertcuykens** [mapbox/mapbox-gl-js#1983]
After map is loaded, sourceObj.setData(traces) doesn't work anymore? It does work in v0.12.3

I haven't a example yet to show but just to let you know already something is wrong in v.0.12.4. I am trying to come up with a example, but it's not straightforward to find why exactly, in complex code that is working in v0.12.3 :)

```js
       traces.features.push({
            "type": "Feature",
            "geometry": {
                "type": "LineString",
                "coordinates": line
            },
            "properties": {
                "id": feed
            }
        })
        sourceObj.setData(traces)
```
![image](https://cloud.githubusercontent.com/assets/863731/12531654/4db9cde0-c200-11e5-83dd-5eb6bec98c12.png)

**Comment by averas** [comment]
Seeing the same..

**Comment by lucaswoj** [comment]
Almost certainly related to https://github.com/mapbox/mapbox-gl-js/commit/b24b6a17a14c3acc4657eae4eb150445cce29902#diff-b170f1c77c7b2cd7bd764fb11a521e3dL21. We should make `Tile#getElementGroups` inspect `Tile#buffers` too.

**Comment by lucaswoj** [comment]
Fixed by #1987
